### PR TITLE
Move SumAggregate.h to functions/lib to allow for reuse

### PR DESCRIPTION
**Pull request by liujiayi771**
SUM(DECIMAL) signature in Spark is different from Presto. The rest of the input
types can re-use Presto's implementation. 

Spark's SUM(DECIMAL) is being added in#5372.

### CODE REVIEW

**Comment by liujiayi771** [comment]
Hi @mbasmanova, as https://github.com/facebookincubator/velox/pull/5372#discussion_r1321507437 discussed, sparksql needs to separately register all types of agg functions of sum, so we need the implementation of sum agg to be placed in `lib/aggregates`. This PR moves sum agg to the lib directory, and only remains the registration-related code in `prestosql/aggregates` and `sparksql/aggregates`. The decimal type has been deleted from sparksql registration (because it is not actually suitable). It will be added in the PR of decimal sum later, and it will add some decimal sum tests for sparksql.

File: velox/functions/lib/aggregates/SumAggregateBase.h [hunk 1]
```diff
@@ -16,22 +16,22 @@
 #pragma once
 
 #include "velox/expression/FunctionSignature.h"
+#include "velox/functions/lib/CheckedArithmeticImpl.h"
 #include "velox/functions/lib/aggregates/DecimalAggregate.h"
 #include "velox/functions/lib/aggregates/SimpleNumericAggregate.h"
-#include "velox/functions/prestosql/CheckedArithmeticImpl.h"
 
-using namespace facebook::velox::functions::aggregate;
+using namespace facebook::velox::aggregate;
```
Comment (mbasmanova) [inline velox/functions/lib/aggregates/SumAggregateBase.h:23]: do not use "using" in header file; looks like this is a pre-existing issue; would you submit a separate PR to fix that, than rebase this one?

File: velox/functions/lib/aggregates/SumAggregateBase.h [hunk 2]
```diff
@@ -16,22 +16,22 @@
...
 
-namespace facebook::velox::aggregate::prestosql {
+namespace facebook::velox::functions::aggregate {
```
Comment (mbasmanova) [inline velox/functions/lib/aggregates/SumAggregateBase.h:25]: Line 199 needs updating as well

File: velox/functions/sparksql/aggregates/SumAggregate.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+#pragma once
+
+#include <string>
+
+#include "velox/exec/AggregateUtil.h"
+
+namespace facebook::velox::functions::aggregate::sparksql {
+
+exec::AggregateRegistrationResult registerSum(const std::string& name);
```
Comment (mbasmanova) [inline velox/functions/sparksql/aggregates/SumAggregate.h:25]: return value is not used; let's change this function to void and remove #include "velox/exec/AggregateUtil.h"

File: velox/functions/lib/aggregates/SumAggregateBase.h (hunk 1 above)
Comment (liujiayi771) [reply velox/functions/lib/aggregates/SumAggregateBase.h]: Hi @mbasmanova, after moving SumAggregateBase.h to `lib`, only SumHook came from another namespace, so I directly removed the using namespace in this PR and modified [SumHook](https://github.com/facebookincubator/velox/pull/6889/files#diff-3d23b55eb5bd17dafffdc46c9674a8c9968a0b26c2d411829d564574d03d549cR132) to use the complete namespace.

File: velox/functions/lib/aggregates/SumAggregateBase.h (hunk 2 above)
Comment (liujiayi771) [reply velox/functions/lib/aggregates/SumAggregateBase.h:23]: Fixed.

**Review by mbasmanova — APPROVED** [review]
Thank you. Looks good.
